core/state/statedb.go

```diff
@@ -933,6 +934,19 @@ func (s *StateDB) Commit(deleteEmptyObjects bool) (common.Hash, error) {
 				storageTrieNodesUpdated += updates
 				storageTrieNodesDeleted += deleted
 			}
+		} else {
```

Comment:
What happens if a state object is deleted and then recreated in the same block / transaction? Won't that end up with missing deleting the old trie nodes?

Also, can it happen that we delete the same contract and recreate it multiple times in a transaction and instead of deleting the itermediate storage slots, we keep deleting the originals already done? Unsure if this can happen post-homestead (no intermediate root). Perhaps before?

Reply:
First of all, `Commit` is only called 
- at end of block post-byzantium, and 
- between each tx, pre-byzantium. 

Now, 
- state object is deleted and then recreated in the same transaction
   - This cannot happen. The object is not deleted until _after_ the transaction ends. This goes for both pre- and post- byz. 
- state object is deleted and then recreated in the same block
   - Yes, this is possible.

Reply:
> What happens if a state object is deleted and then recreated in the same block / transaction? Won't that end up with missing deleting the old trie nodes?

I think you are right. It can be possible before byzantium... 

> Also, can it happen that we delete the same contract and recreate it multiple times in a transaction 

Is it possible that destruction and creation can happen in a single transaction? Not sure about it.

Reply:
> Is it possible that destruction and creation can happen in a single transaction? Not sure about it.

No. The answer is no.